src/substrait/textplan/converter/PlanPrinterVisitor.cpp

```diff
@@ -442,10 +442,28 @@ std::any PlanPrinterVisitor::visitWindowFunction(
 
 std::any PlanPrinterVisitor::visitIfThen(
     const ::substrait::proto::Expression::IfThen& ifthen) {
-  errorListener_->addError(
-      "If then expressions are not yet supported: " +
-      ifthen.ShortDebugString());
-  return std::string("IFTHEN_NOT_YET_IMPLEMENTED");
+  std::stringstream text;
+  text << "IFTHEN(";
+  bool hasPreviousText = false;
+  for (const auto& clause : ifthen.ifs()) {
+    if (hasPreviousText) {
+      text << ", ";
+    }
+    text << ANY_CAST(std::string, visitExpression(clause.if_()));
+    text << ", ";
+    text << ANY_CAST(std::string, visitExpression(clause.then()));
```

Comment:
What happens if `then` is not set?  Do we crash?  It's ok if it's invalid.

Reply:
Added check and test for the same.